File: src-gen/DartSim.cpp/src/SimulationParams.cc

#include "SimulationParams.hh"

#include "umlrtobjectclass.hh"
#include "umlrtobjectclassgeneric.hh"
#include <cstddef>

const UMLRTObject_field SimulationParams::fields[] = 
{
};
SimulationParams::SimulationParams()
{
    mapSize = ( 40 );
    squareMap = ( false );
    altitudeLevels = ( 4 );
    changeAltitudeLatencyPeriods = ( 1 );
    optimalityTest = ( false );
}











const UMLRTObject_class UMLRTType_SimulationParams = 
{
    UMLRTObjectInitialize<SimulationParams>,
    UMLRTObjectCopy<SimulationParams>,
    UMLRTObject_decode,
    UMLRTObject_encode,
    UMLRTObjectDestroy<SimulationParams>,
    UMLRTObject_fprintf,
    "SimulationParams",
    NULL,
    {
        sizeof( SimulationParams ),
        0,
        SimulationParams::fields
    },
    UMLRTOBJECTCLASS_DEFAULT_VERSION,
    UMLRTOBJECTCLASS_DEFAULT_BACKWARDS
};

File: src-gen/DartSim.js/SimulationParams.js
module.exports = class SimulationParams {
	constructor() {
		this.mapSize = 40,
		this.squareMap = false,
		this.altitudeLevels = 4,
		this.changeAltitudeLatencyPeriods = 1,
		this.optimalityTest = false,
		this.longRangeSensor = null,
		this.downwardLookingSensor = null,
		this.threat = null
	}
}